factory/alchemy.py

```diff
@@ -91,6 +91,8 @@ def _create(cls, model_class, *args, **kwargs):
             session_persistence = SESSION_PERSISTENCE_FLUSH
 
         obj = model_class(*args, **kwargs)
+        if session is None:
+            raise RuntimeError('No session provided.')
```

Comment:
Nit: Do you mind using double quotes for this message? 

Both myself and @rbarrois prefer double quotes for strings intended for human consumption, and single-quotes for machine consumption.